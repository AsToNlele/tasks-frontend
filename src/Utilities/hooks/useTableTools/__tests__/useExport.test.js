import { act, renderHook } from '@testing-library/react-hooks';
import columns from './__fixtures__/columns.fixtures';
import items from './__fixtures__/items.fixtures';
import useExport, { jsonForItems, csvForItems } from '../useExport';

describe('useExport', () => {
  const exporter = jest.fn(() => Promise.resolve(items));
  const defaultOptions = {
    exporter,
    columns,
  };

  it('returns an export config toolbar config', () => {
    const { result } = renderHook(() => useExport(defaultOptions));
    expect(result.current.toolbarProps.exportConfig).toBeDefined();
    expect(result).toMatchSnapshot();
  });

  it('returns an export config toolbar config', () => {
    const { result } = renderHook(() =>
      useExport({
        ...defaultOptions,
        isDisabled: true,
      })
    );
    expect(result.current.toolbarProps.exportConfig.isDisabled).toBe(true);
  });

  it('calls the exporter via onSelect', () => {
    const { result } = renderHook(() => useExport(defaultOptions));

    act(() => {
      result.current.toolbarProps.exportConfig.onSelect(null, 'csv');
    });

    expect(exporter).toHaveBeenCalled();

    act(() => {
      result.current.toolbarProps.exportConfig.onSelect(null, 'json');
    });

    expect(exporter).toHaveBeenCalled();
  });
});

describe('jsonForItems', () => {
  it('returns an json export of items', () => {
    const result = jsonForItems({
      columns,
      items: items,
    });

    expect(result).toMatchSnapshot();
  });
});

describe('csvForItems', () => {
  it('returns an csv export of items', () => {
    const result = csvForItems({
      columns,
      items: items,
    });

    expect(result).toMatchSnapshot();
  });
});
