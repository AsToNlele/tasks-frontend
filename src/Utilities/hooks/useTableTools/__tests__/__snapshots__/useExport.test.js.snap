// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csvForItems returns an csv export of items 1`] = `"data:text/csv;charset=utf-8,Task,Systems,Run%20date/time%0A%22%5Bobject%20Object%5D%22,%22%5Bobject%20Object%5D%22,%22%5Bobject%20Object%5D%22%0A%22%5Bobject%20Object%5D%22,%22%5Bobject%20Object%5D%22,%22%5Bobject%20Object%5D%22"`;

exports[`jsonForItems returns an json export of items 1`] = `"data:application/json;charset=utf-8,%5B%7B%22Task%22:%7B%22title%22:%22taskA%22,%22id%22:1,%22start%22:%222022-04-21T10:10:00%22,%22end%22:null,%22initiated_by%22:%22Michael%22,%22status%22:%22running%22,%22system_count%22:10%7D,%22Systems%22:%7B%22title%22:%22taskA%22,%22id%22:1,%22start%22:%222022-04-21T10:10:00%22,%22end%22:null,%22initiated_by%22:%22Michael%22,%22status%22:%22running%22,%22system_count%22:10%7D,%22RunDateTime%22:%7B%22title%22:%22taskA%22,%22id%22:1,%22start%22:%222022-04-21T10:10:00%22,%22end%22:null,%22initiated_by%22:%22Michael%22,%22status%22:%22running%22,%22system_count%22:10%7D%7D,%7B%22Task%22:%7B%22title%22:%22taskB%22,%22id%22:2,%22start%22:%222022-04-20T10:10:00%22,%22end%22:%222022-04-20T11:10:00%22,%22initiated_by%22:%22Michael%22,%22status%22:%22complete%22,%22system_count%22:5%7D,%22Systems%22:%7B%22title%22:%22taskB%22,%22id%22:2,%22start%22:%222022-04-20T10:10:00%22,%22end%22:%222022-04-20T11:10:00%22,%22initiated_by%22:%22Michael%22,%22status%22:%22complete%22,%22system_count%22:5%7D,%22RunDateTime%22:%7B%22title%22:%22taskB%22,%22id%22:2,%22start%22:%222022-04-20T10:10:00%22,%22end%22:%222022-04-20T11:10:00%22,%22initiated_by%22:%22Michael%22,%22status%22:%22complete%22,%22system_count%22:5%7D%7D%5D"`;

exports[`useExport returns an export config toolbar config 1`] = `
Object {
  "all": Array [
    Object {
      "toolbarProps": Object {
        "exportConfig": Object {
          "isDisabled": false,
          "onSelect": [Function],
        },
      },
    },
  ],
  "current": Object {
    "toolbarProps": Object {
      "exportConfig": Object {
        "isDisabled": false,
        "onSelect": [Function],
      },
    },
  },
  "error": undefined,
}
`;
